content/blog/quine-quine/index.md

```diff
@@ -0,0 +1,227 @@
+---
+title: Writing a program that writes itself
+date: "2019-11-04T12:00:00.000Z"
+description: What is a Quine and how to write one.
+---
+
+Recursion is one of those concepts in programming which have always been
+incredibly fascinating to me. _What do you mean that I can call the
+function from the function itself?_
+
+Well, there is a class of programs that bring the whole idea of recursion
+to a completely different level, which is programs that are able to
+replicate themselves. These programs, named **Quines**, produce a copy of
+their own source code as their only output.
+
+![Drawing Hands, M.C.Escher](./escher.jpg)
+
+The name "quine" was introduced by Douglas Hofstadter in _Gödel, Escher,
+Bach: An Eternal Golden Braid_ in honor of [Willard Van Orman
+Quine](https://en.wikipedia.org/wiki/Willard_Van_Orman_Quine), an American
+analytic philosopher who made extensive studies in the field of
+self-reference. There is a paradox named after him that goes:
+
+> "Yields falsehood when preceded by its quotation" yields falsehood when preceded by its quotation.
+
+This is a more elaborate version of the [Liar
+paradox](https://en.wikipedia.org/wiki/Liar_paradox), where the sentence
+`Yields falsehood when preceded by its quotation`, when preceded by itself
+in quotation marks, gives a new expression. If this new expression is true,
+then it says of itself that it must be false. But if it's false, then what
+it says of itself must be true.
+
+Pretty puzzling, huh?
+
+## Some rules
+
+Before we start and try to write a Quine, we have to define some ground
```

Comment:
~Same here, re. capitalisation, and in other places in this post.~

I just read [this](https://www.grammarly.com/blog/capitalization-rules/) to check, and it kind of confirms what I said, but quine/Quine is named after a person, so should we capitalise it anyway?

Wikipedia's [list of inventions named after people](https://en.wikipedia.org/wiki/List_of_inventions_named_after_people) shows that it's a mixed bag. For example, bloomers and derrick and guillotine are not capitalised, but most are.

Wikipedia does not capitalise in [its article about quines](https://en.wikipedia.org/wiki/Quine_(computing)).

How did Douglas Hofstadter capitalise it?

I think the answer is: I don't know any more. My guess that Quine, capitalised, is just fine.